src/tests/test_utils/functional_test_utils/layer_tests_summary/run_parallel.py

```diff
@@ -282,156 +298,143 @@ def __get_test_list_by_runtime(self):
             exit(0)
-        return test_list
+        return tests_dict
 
     def __get_test_list_by_cache(self):
-        test_list_cache = list()
+        tests_dict_cache = dict()
         if os.path.isfile(self._cache_path):
             logger.info(f"Get test list from cache file: {self._cache_path}")
             with open(self._cache_path, "r") as cache_file:
                 for line in cache_file.readlines():
                     pos = line.find(":")
-                    time = line[:pos]
-                    test_name = line[pos+1:]
-                    if not constants.DISABLED_PREFIX in test_name:
-                        test_list_cache.append(TestStructure(test_name.replace("\n", ""), time))
-        logger.info(f"Len test_list_cache: {len(test_list_cache)}")
-        return test_list_cache
-
-    def __generate_test_lists(self, test_list_cache: list, test_list_runtime:list):
-        cached_test_list = list()
-        runtime_test_test = list()
-        cached_test_list_names = list()
-
-        for test in test_list_cache:
-            if test._name in test_list_runtime and not test in self._excluded_tests_re:
-                cached_test_list.append(test)
-                cached_test_list_names.append(test._name)
-        for test in test_list_runtime:
-            if not test in cached_test_list_names and not test in self._excluded_tests_re:
-                runtime_test_test.append(test)
-
-        if len(runtime_test_test) > 0:
+                    time = int(line[:pos])
+                    test_name = line[pos+1:].replace("\n", "")
+                    test_suite = test_name[:test_name.find(".")]
+
+                    if (self._split_unit == "test"):
+                        if constants.DISABLED_PREFIX not in test_name:
+                            if (time != -1):
+                                tests_dict_cache[test_name] = tests_dict_cache.get(test_name, 0) + time
+                    elif (self._split_unit == "suite"):
+                        if constants.DISABLED_PREFIX not in test_suite:
+                            if (time == -1):
+                                tests_dict_cache[test_suite] = tests_dict_cache.get(test_suite, -1)
+                            else:
+                                tests_dict_cache[test_suite] = tests_dict_cache.get(test_suite, 0) + time
+
+        logger.info(f"Len tests_dict_cache: {len(tests_dict_cache)}")
+        return tests_dict_cache
+
+    def __generate_test_lists(self, test_dict_cache: dict, test_dict_runtime: dict):
+        cached_test_dict = dict()
+        runtime_test_dict = dict()
+
+        for test in test_dict_cache:
+            if test in test_dict_runtime:
+                cached_test_dict[test] = test_dict_cache[test] if test_dict_cache[test] != -1 else test_dict_runtime.get(test, -1)
+
+        for test in test_dict_runtime:
+            if test not in cached_test_dict:
```

Comment:
Please, check the merge!!!! Check expected failures

Reply:
yes, sorry, missed that string during the merge. Reverted